ArduPlane/ArduPlane.cpp

```diff
@@ -963,6 +963,25 @@ bool Plane::flight_option_enabled(FlightOptions flight_option) const
     return g2.flight_options & flight_option;
 }
 
+// This will only apply in Guided mode, other modes ignore it
+// all the checks are in the Guided mode code, this is _desired_ speed
+bool Plane::set_desired_speed(float speed)
```

Comment:
Looks like this would spam the GCS with speed change messages, I'm calling this at 10Hz. It also doesn't set target_airspeed_accel, which (I think) I want.

Reply:
Fair enough, in that case this method should be called `set_desired_airspeed`. You also need to duplicate the range check for min/max airspeeds per `do_change_speed`.

Reply:
I think its fine for this to work in all modes, so you would set the brilliantly named `new_airspeed_cm`.

Reply:
> Fair enough, in that case this method should be called `set_desired_airspeed`. You also need to duplicate the range check for min/max airspeeds per `do_change_speed`.

That's why I was calling calc_airspeed_errors() it was doing that and a whole lot more. I'm thinking if calc_airspeed_errors() is being called 10Hz- do I need to do all those checks?

Reply:
> Fair enough, in that case this method should be called `set_desired_airspeed`. You also need to duplicate the range check for min/max airspeeds per `do_change_speed`.

I would agree with you, but I used set_desired_speed because it's an existing virtual function from AP_Vehicle that is already implemented in Copter.